rc/lsp.kak

```diff
@@ -452,19 +452,19 @@ Jump to the next or previous diagnostic error" %{
         current=""
         prev=""
         selection=""
-        for e in $@; do
-            if [[ $e == *"$errorCompare"* ]]; then
+        for e in "$@"; do
+            if [ "${e#*$errorCompare}" != "$e" ]; then # e contains errorCompare
                 e=${e%,*}
                 line=${e%.*}
                 column=${e#*.}
-                if [[ $line -eq $kak_cursor_line && $column -eq $kak_cursor_column ]]; then
+                if [ "$line" -eq "$kak_cursor_line" ] && [ "$column" -eq "$kak_cursor_column" ]; then
```

Comment:
Quotes should not be needed.

Reply:
Well, less characters...  Also, quotes should only be needed for strings containing whitespaces; AFAIK, the variables involved here don't have whitespaces, so quotes shouldn't be needed.

Reply:
Remember: all variables are strings in a POSIX shell.  One shouldn't need to quote them as one does in C, Go or other programming languages, *unless* those strings carry some special symbol that could (or could not, depends of your intent) be erroneously interpreted by the shell.